app/import_functions.go

```diff
@@ -960,123 +982,219 @@ func (a *App) importAttachment(data *AttachmentImportData, post *model.Post, tea
 			return err
 		}
-	} else {
-		if _, err = a.Srv.Store.Post().Overwrite(post); err != nil {
+	}
+
+	for _, post := range postsForOverwriteList {
+		if _, err := a.Srv.Store.Post().Overwrite(post); err != nil {
 			return err
 		}
 	}
 
-	if data.FlaggedBy != nil {
-		var preferences model.Preferences
+	var lastPostWithData *postAndData
+	repliesBulk := []ReplyImportData{}
+	for _, postWithData := range postsWithData {
+		if postWithData.postData.FlaggedBy != nil {
+			var preferences model.Preferences
 
-		for _, username := range *data.FlaggedBy {
-			var user *model.User
-			user, err = a.Srv.Store.User().GetByUsername(username)
-			if err != nil {
-				return model.NewAppError("BulkImport", "app.import.import_post.user_not_found.error", map[string]interface{}{"Username": username}, err.Error(), http.StatusBadRequest)
+			for _, username := range *postWithData.postData.FlaggedBy {
+				user := users[username]
+
+				preferences = append(preferences, model.Preference{
+					UserId:   user.Id,
+					Category: model.PREFERENCE_CATEGORY_FLAGGED_POST,
+					Name:     postWithData.post.Id,
+					Value:    "true",
+				})
 			}
 
-			preferences = append(preferences, model.Preference{
-				UserId:   user.Id,
-				Category: model.PREFERENCE_CATEGORY_FLAGGED_POST,
-				Name:     post.Id,
-				Value:    "true",
-			})
+			if len(preferences) > 0 {
+				if err := a.Srv.Store.Preference().Save(&preferences); err != nil {
+					return model.NewAppError("BulkImport", "app.import.import_post.save_preferences.error", nil, err.Error(), http.StatusInternalServerError)
+				}
+			}
 		}
 
-		if len(preferences) > 0 {
-			if err := a.Srv.Store.Preference().Save(&preferences); err != nil {
-				return model.NewAppError("BulkImport", "app.import.import_post.save_preferences.error", nil, err.Error(), http.StatusInternalServerError)
+		if postWithData.postData.Reactions != nil {
+			for _, reaction := range *postWithData.postData.Reactions {
+				if err := a.importReaction(&reaction, postWithData.post, dryRun); err != nil {
+					return err
+				}
 			}
 		}
-	}
 
-	if data.Reactions != nil {
-		for _, reaction := range *data.Reactions {
-			if err := a.importReaction(&reaction, post, dryRun); err != nil {
-				return err
+		if postWithData.postData.Replies != nil {
+			repliesBulk = append(repliesBulk, *postWithData.postData.Replies...)
+			if len(repliesBulk) >= importMultiplePostsThreshold {
+				err := a.importReplies(repliesBulk, postWithData.post, postWithData.team.Id, dryRun)
+				if err != nil {
+					return err
+				}
+				repliesBulk = []ReplyImportData{}
 			}
 		}
+		a.updateFileInfoWithPostId(postWithData.post)
+		lastPostWithData = &postWithData
 	}
 
-	if data.Replies != nil {
-		for _, reply := range *data.Replies {
-			if err := a.importReply(&reply, post, team.Id, dryRun); err != nil {
-				return err
-			}
+	if len(repliesBulk) >= importMultiplePostsThreshold && lastPostWithData != nil {
```

Comment:
I'm probably missing something, but why is there a threshold check in this condition? What happens if it doesn't pass.

Reply:
Actually, this is wrong, it is for the remaning replies, so should be executed if the repliesBulk is greater than 0